app/api/services/suppliers.py

```diff
@@ -25,3 +25,11 @@ def get_supplier_by_code(self, code):
         )
             .filter(Supplier.code == code)
             .one_or_none())
+
+    def get_suppliers_by_name_keyword(self, keyword):
+        return (db.session.query(Supplier)
+                .filter(Supplier.name.ilike('%{}%'.format(keyword)))
+                .filter(Supplier.status != 'deleted')
+                .order_by(Supplier.name.asc())
+                .options(joinedload(Supplier.frameworks), noload('frameworks.framework.lots'))
```

Comment:
This gets rid of the N+1 errors on the `SupplierFrameworks` relationship, and the unused load error on the `FrameworkLots` relationship.

Reply:
since this is a new function, it might be an idea here to add `raiseload('*')` to prevent further attempts in the future.
http://docs.sqlalchemy.org/en/latest/orm/loading_relationships.html#prevent-lazy-with-raiseload

or do the mapping to a dict here and return that, to prevent further loading of supplier related tables. i.e. `return [s._asdict() for s in supplier]`